### PR TITLE
Citlally

### PR DESCRIPTION
**Pull request by Cytla39**
## Release Notes

- **New Features**
  - Introduced multiple API endpoints for managing categories, clients, employees, menus, tables, orders, products, and reservations, supporting CRUD operations.
  - Added a new `Producto` and `Menu` model to enhance product and menu management.
  
- **Improvements**
  - Enhanced error handling for JWT verification to improve security and debugging.
  - Updated the database schema with new tables and relationships to support a more complex data structure.

- **Chores**
  - Updated development dependencies for improved type definitions. 
  - Modified `.gitignore` to exclude sensitive environment files.

### CODE REVIEW

**Comment by FGArandaRodriguez** [comment]
@coderabbit review

File: pages/api/menu.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            const menu = await prisma.menu.findMany();
+            res.status(200).json(menu);
+        }catch(error){
+            res.status(500).json({message:'Error al obtener el  Menu', error});
+        }
+    } else if (req.method === 'POST'){
+        const { nombre, descripcion, precio, imagen, idCategoria } = req.body;
+        if (!nombre || !precio || !descripcion || !imagen || !idCategoria){
+            return res.status(400).json({message: `Faltan campos requeridos ${nombre}`})
+        }
```
[the remark this answers is not in the document]
Comment (Cytla39) [reply pages/api/menu.ts]: Screenshot 2024-10-30 125135

File: prisma/migrations/20241029203854_init/migration.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/*
+  Warnings:
+
+  - You are about to drop the `categoria` table. If the table is not empty, all the data it contains will be lost.
+
+*/
```
[the remark this answers is not in the document]
Comment (Cytla39) [reply prisma/migrations/20241029203854_init/migration.sql:6]: Screenshot 2024-10-30 123127

File: prisma/migrations/20241025055451_init/migration.sql [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/*
+  Warnings:
+
+  - Added the required column `id_categoria` to the `Menu` table without a default value. This is not possible if the table is not empty.
+
+*/
```
[the remark this answers is not in the document]
Comment (Cytla39) [reply prisma/migrations/20241025055451_init/migration.sql:6]: Screenshot 2024-10-30 124029

File: prisma/migrations/20241025055451_init/migration.sql [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+-- AlterTable
+ALTER TABLE `menu` ADD COLUMN `id_categoria` INTEGER NOT NULL;
+
+-- CreateTable
+CREATE TABLE `Categoria` (
+    `id` INTEGER NOT NULL AUTO_INCREMENT,
+    `nombre` VARCHAR(191) NOT NULL,
+
+    PRIMARY KEY (`id`)
+) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
```
[the remark this answers is not in the document]
Comment (Cytla39) [reply prisma/migrations/20241025055451_init/migration.sql]: Screenshot 2024-10-30 123127

File: pages/api/menu.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                    descripcion,
+                    precio: parseFloat(precio),
+                    imagen,
+                    id_categoria: idCategoria
+                },
+            });
+            res.status(201).json(newMenu);
+        }catch(error){
+            res.status(500).json({message: 'Error al actualizar el Menu', error});
+        }
```
[the remark this answers is not in the document]
Comment (Cytla39) [reply pages/api/menu.ts]: image

File: pages/api/menu.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    }else if(req.method === 'PUT'){
+        const { id, nombre, descripcion, precio, imagen, idCategoria } = req.body;
+        if (!nombre || !precio || !descripcion || !imagen || !idCategoria){
+            return res.status(400).json({message: 'Faltan campos requeridos'})
+        }
```
[the remark this answers is not in the document]
Comment (Cytla39) [reply pages/api/menu.ts]: Screenshot 2024-10-30 125244

File: pages/api/menu.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+export default async function name(
+    req: NextApiRequest,
+    res: NextApiResponse
+) {
```
[the remark this answers is not in the document]
Comment (Cytla39) [reply pages/api/menu.ts]: Screenshot 2024-10-30 125404

**Pull request merged**